doc/source/programs/gdal_raster_reproject.rst

```diff
@@ -129,6 +162,106 @@ Standard options
     Alignment means that xmin / resx, ymin / resy,
     xmax / resx and ymax / resy are integer values.
 
+Advanced options
+++++++++++++++++
+
+.. option:: --srcnodata <SRCNODATA>
+
+    Set nodata masking values for input bands (different values can be supplied
+    for each band). If more than one value is supplied all values should be quoted
+    to keep them together as a single operating system argument.
+    Masked values will not be used in interpolation (details given in :ref:`gdalwarp_nodata`)
+
+    Use a value of ``None`` to ignore intrinsic nodata settings on the source dataset.
+
+    When this option is set to a non-``None`` value, it causes the ``UNIFIED_SRC_NODATA``
+    warping option (see :cpp:member:`GDALWarpOptions::papszWarpOptions`) to be
+    set to ``YES``, if it is not explicitly set.
+
+    If ``--srcnodata`` is not explicitly set, but the source dataset has nodata values,
+    they will be taken into account, with ``UNIFIED_SRC_NODATA`` at ``PARTIAL``
+    by default.
+
+.. option:: --dstnodata <DSTNODATA>
+
+    Set nodata values
+    for output bands (different values can be supplied for each band).  If more
+    than one value is supplied all values should be quoted to keep them together
+    as a single operating system argument.  New files will be initialized to this
+    value and if possible the nodata value will be recorded in the output
+    file. Use a value of ``None`` to ensure that nodata is not defined.
+    If this argument is not used then nodata values will be copied from the source dataset.
+
+.. option:: --addalpha
+
+    Create an output alpha band to identify nodata (unset/transparent) pixels.
+
```

Comment:
Maybe mention value 0 will be written to alpha band to mean nodata, and 255 to mean data when the output datatype is "byte". I do not know how it behaves with other datatypes.

Reply:
note added